engines/xmlengine.py

```diff
@@ -124,7 +124,7 @@ def table_exists(self, dbname, tablename):
 
     def to_csv(self):
         """Export table from xml engine to CSV file"""
-        keys = [columnname[0] for columnname in self.table.columns]
+        keys = [columnname for columnname in self.table.get_insert_columns(join=False, create=True)]
```

Comment:
Is the list comprehension doing anything now? It seems like `get_insert_columns(join=False)` will return a list and the list comprehension will just take the items from that list and use them to make a new list.